docs/dev-guide/contributing/branching.rst

```diff
@@ -23,47 +23,47 @@ master
 
 This is the latest bleeding-edge code. All new feature work should be done out
 of this branch. Typically this is the development branch for future, unreleased
-"x" or "x.y" release.
+``x`` or ``x.y`` release.
 
 
-Release Branches
-----------------
+Version-Specific Branches
+-------------------------
 
-A branch will be made for each x.y release stream named "x.y-release". For example,
-the 2.4 release lives in a branch named "2.4-release". Increments of "z" releases
-occur within the same release branch and are identified by tags.
+Each ``x.y`` release will have one corresponding branch called ``x.y-dev``. For
+example, all work for the 2.7.z series of releases gets merged into ``2.7-dev``.
 
-The HEAD of each release branch points to a tagged release version. When a new
-"z" increment version of Pulp is released, the testing branch is merged
-into the release branch and the new HEAD of the release branch is tagged.
-Development occurs on a separate development branch.
 
-Release branches are where Read The Docs builds from, so in some situations
-documentation commits may be merged into a release branch after a release has
-occurred. For example if a known issue is discovered in a release after it is
-released, it may be added to the release notes. In those situations the
-release tag will stay the same and diverge from HEAD.
+Build Tags
+----------
 
+Builds will be represented only as tags.
 
-Testing Branches
-----------------
+.. note:: In the past, the latest beta and GA release of an x.y stream would be
+    represented additionally by branches, but that is no longer the case as of
+    pulp 2.7.
 
-Each x.y release will also have a branch for testing builds named "x.y-testing". For example, the
-2.4 stream has a "2.4-testing" branch. This branch is made when we are ready to begin regression
-testing 2.4.1. After 2.4.0 has been released, the 2.4-dev branch will be merged into 2.4-testing,
-and this branch will be used to make beta builds. Release candidates will also be built out of this
-branch. Once we believe the 2.4-testing branch has code that is ready to be release, it will be
-merged into 2.4-release.
 
+Build Lifecycle
+---------------
 
-Development Branches
---------------------
+Beta releases will be built from the tip of an ``x.y-dev`` branch. If the beta
```

Comment:
I updated this paragraph to mention alphas and betas. But, since this particular document is about our "Branching Model", I don't think it's the right place to define alpha vs. beta. I think that would be best defined in the user docs, with a complete description of our versioning approach.